main.go

```diff
@@ -64,56 +64,61 @@ func isKnownStateVersion(statesVersions map[string][]string, versionID, path str
 }
 
 // Refresh the DB
-// This should be the only direct bridge between the state provider and the DB
-func refreshDB(syncInterval uint16, d *db.Database, sp state.Provider) {
+// This should be the only direct bridge between the state providers and the DB
+func refreshDB(syncInterval uint16, d *db.Database, sps []state.Provider) {
 	interval := time.Duration(syncInterval) * time.Minute
+	log.Debugf("Providers: %+v\n", sps)
 	for {
 		log.Infof("Refreshing DB")
-		states, err := sp.GetStates()
-		if err != nil {
-			log.WithFields(log.Fields{
-				"error": err,
-			}).Error("Failed to retrieve states. Retrying in 1 minute.")
-			time.Sleep(interval)
-			continue
-		}
+		log.Debugf("Total providers: %d\n", len(sps))
+		for i, sp := range sps {
```

Comment:
Looping inside `refreshDB()` means each provider will be parsed one a time. Maybe we could parallelize providers instead by looping ouside the call to `refreshDB()` in `main()` and keep `refreshDB()` the way it is.

Reply:
Great idea that I should have thought of! 
This will allow to not interrupt all the fetching processes if one of them has an error and reduce the time that the refresh takes
I will do that